crates/accelerate/src/target_transpiler/mod.rs

```diff
@@ -0,0 +1,1285 @@
+                    search_set.insert(vec);
+                }
+            }
+        }
+        let mut incomplete_basis_gates: Vec<String> = vec![];
+        let mut size_dict: IndexMap<usize, usize, RandomState> = IndexMap::default();
+        *size_dict
+            .entry(1)
+            .or_insert(self.num_qubits.unwrap_or_default()) = self.num_qubits.unwrap_or_default();
+        for qarg in &search_set {
+            if qarg.len() == 1 {
+                continue;
+            }
+            *size_dict.entry(qarg.len()).or_insert(0) += 1;
+        }
+        for (inst, qargs_props) in self.gate_map.iter() {
+            let mut qarg_len = qargs_props.len();
+            let qargs_keys: IndexSet<&Option<Qargs>> = qargs_props.keys().collect();
+            let qarg_sample = qargs_keys.iter().next().cloned();
+            if let Some(qarg_sample) = qarg_sample {
+                if !strict_direction {
+                    let mut qarg_set: HashSet<SmallVec<[PhysicalQubit; 2]>, RandomState> =
+                        HashSet::default();
+                    for qarg in qargs_keys {
+                        let mut qarg_set_vec: Qargs = smallvec![];
+                        if let Some(qarg) = qarg {
+                            let mut to_vec = qarg.to_owned();
+                            to_vec.sort_unstable();
+                            qarg_set_vec = to_vec;
+                        }
+                        qarg_set.insert(qarg_set_vec);
+                    }
+                    qarg_len = qarg_set.len();
+                }
+                if let Some(qarg_sample) = qarg_sample {
+                    if qarg_len != *size_dict.entry(qarg_sample.len()).or_insert(0) {
+                        incomplete_basis_gates.push(inst.to_owned());
+                    }
+                }
+            }
+        }
+        if strict_direction {
+            self.non_global_strict_basis = Some(incomplete_basis_gates);
+            self.non_global_strict_basis.as_ref().unwrap()
+        } else {
+            self.non_global_basis = Some(incomplete_basis_gates.to_owned());
+            self.non_global_basis.as_ref().unwrap()
+        }
+    }
+
+    /// Get all non_global operation names.
+    pub fn get_non_global_op_names(&mut self, strict_direction: bool) -> Option<&[String]> {
+        if strict_direction {
+            if self.non_global_strict_basis.is_some() {
+                return self.non_global_strict_basis.as_deref();
+            }
+        } else if self.non_global_basis.is_some() {
+            return self.non_global_basis.as_deref();
+        }
+        return Some(self.generate_non_global_op_names(strict_direction));
+    }
+
+    /// Gets all the operation names that use these qargs. Rust native equivalent of ``BaseTarget.operation_names_for_qargs()``
+    pub fn operation_names_for_qargs(
+        &self,
+        qargs: &Option<Qargs>,
+    ) -> Result<HashSet<&String, RandomState>, TargetKeyError> {
```

Comment:
I would use `&str` here instead of `&String` also if you're using hashbrown you don't need to set `RandomState` since that's the default (which is a big part of why we use it).